apps/modernization-ui/src/apps/page-builder/pages/ManagePages/ManagePages.tsx

```diff
@@ -1,20 +1,45 @@
-import { PageProvider } from 'page';
-import { useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
 import { PageBuilder } from '../PageBuilder/PageBuilder';
 import './ManagePages.scss';
-import { ManagePagesTableWrapper } from './ManagePagesTableWrapper';
+import { PagesContext } from 'apps/page-builder/context/PagesContext';
+import { fetchPageSummaries } from './usePageSummaryAPI';
+import { ManagePagesTable } from './ManagePagesTable';
+import { UserContext } from 'user';
 
 export const ManagePages = () => {
-    const [pageSize] = useState(10);
+    const [pages, setPages] = useState(null);
+    const { sortBy, sortDirection, currentPage, pageSize } = useContext(PagesContext);
+    const [searchString] = useState('');
+    const { state } = useContext(UserContext);
+    const token = `Bearer ${state.getToken()}`;
+    const [isLoading, setIsLoading] = useState(true);
+    const [totalElements, setTotalElements] = useState(0);
+
+    useEffect(() => {
+        // get Pages
+        fetchPageSummaries(token, searchString, sortBy.toLowerCase() + ',' + sortDirection, currentPage, pageSize).then(
+            (data: any) => {
+                setPages(data.content);
+                setTotalElements(data.totalElements);
+                setIsLoading(false);
+            }
+        );
+    }, [searchString, currentPage, pageSize, sortBy, sortDirection]);
+
     return (
         <>
             <PageBuilder page="manage-pages">
                 <div className="manage-pages">
                     <div className="manage-pages__container">
                         <div className="manage-pages__table">
-                            <PageProvider pageSize={pageSize}>
-                                <ManagePagesTableWrapper />
-                            </PageProvider>
+                            {pages && !isLoading && (
+                                <ManagePagesTable
+                                    summaries={pages}
+                                    currentPage={currentPage}
+                                    pageSize={pageSize}
+                                    totalElements={totalElements}
+                                />
+                            )}
```

Comment:
do we want to hide the table while loading or is there any support for a loading indicator?

Reply:
Added the Spinner